pkg/credentials/provider.go

```diff
@@ -34,6 +35,20 @@ func NewProvider(
 	return &cfg, nil
 }
 
+func createSession(stsEndpoint string) *session.Session {
+	var config *aws.Config
+
+	if len(stsEndpoint) > 0 {
+		config = &aws.Config{
+			Endpoint: aws.String(stsEndpoint),
+		}
+	} else {
+		config = &aws.Config{}
+	}
+
+	return session.Must(session.NewSession(config))
+}
```

Comment:
The session creation logic was getting a bit windy so I extracted it into a helper method. Basically, if the endpoint is defined we spin up a session config object which contains it, otherwise we use an empty config object.